src/pudl/transform/eia923.py

```diff
@@ -1223,3 +1224,59 @@ def _core_eia923__fuel_receipts_costs(
     frc_df = frc_df.dropna(subset=["energy_source_code"])
 
     return frc_df
+
+
+@asset
+def _core_eia923__cooling_system_information(
+    raw_eia923__cooling_system_information: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transforms the eia923__cooling_system_information dataframe.
+
+    Applies typical NA conversion and date conversion.
+
+    As of 2024-02-28: 2008 and 2009 only have annual rates, but the
+    "*_rate_gallons_per_minute" values are otherwise monthly; we leave them NA
+    for 2008-2009 to avoid confusion.
+
+    As of 2024-02-28: In 2008 and 2009 the rate columns are labeled as "0.1
+    cubic feet per second"; In 2013 they change it to gallons/min.
+
+    If taken to mean that the earlier unit is literally deci-cubic-feet per
+    second, we find that all these rates jump 10x when we hit the 2013 data;
+    so we interpret "0.1 cubic feet per second" to mean "cubic feet per
+    second, with precision of 0.1 cfs."
+    """
+    csi_df = raw_eia923__cooling_system_information
+    csi_df = csi_df.pipe(pudl.helpers.fix_eia_na).pipe(pudl.helpers.convert_to_date)
+
+    cfs_in_gpm = 448.8311688
+    cooling_water_rate_cols = (
+        "consumption_rate_gallons_per_minute",
+        "discharge_rate_gallons_per_minute",
+        "withdrawal_rate_gallons_per_minute",
+    )
+    cfs_years_mask = csi_df["report_date"].dt.year.isin(range(2008, 2013))
+    csi_df.loc[cfs_years_mask, cooling_water_rate_cols] *= cfs_in_gpm
+
+    return csi_df.pipe(apply_pudl_dtypes, group="eia923")
+
+
+@asset_check(asset=_core_eia923__cooling_system_information, blocking=True)
+def cooling_system_information_check(csi):
+    """Check data quality.
+
+    - no completely null columns
+    - withdrawal == discharge + consumption at least 90% of the time
```

Comment:
This was a property that seems like it's always true in theory - based on a bunch of spot checks. But 🤷 

Hmm, I should probably pull out that 'slowly varying' logic from above and apply that to the various rates here, too. A TODO for tomorrow.

Reply:
The description and the actual math are slightly different here, would be good to explain this isn't `==` but actually ~99% comparison.